File: rosbasic/script/tf_frame_broadcaster.py
#!/usr/bin/env python

import tf
import rospy

import roslib
roslib.load_manifest('rosbasic')

def tf_frame_broadcaster():
	""" This whole program is all same as static transform publisher ! """

	rospy.init_node("tf_frame_broadcaster_node")
	broadcaster = tf.TransformBroadcaster()
	rate = rospy.Rate(10)

	while not rospy.is_shutdown():

		#Broadcast the transformation from tf frame child to parent on ROS topic "/tf
		translation = (0.0,  -2.0 , 0.0 )
		rotation = 	  (0.0 , 0.0 , 0.0 , 1.0)
		time = rospy.Time(0)
		child = "babe"
		parent = "mother"

		# sendTransform(translation, rotation, time, child, parent)
		# rotation will be [0,0,0,1] matrix in this program because it is Quaternion matrix (rotation may be any rotational matrix)
		broadcaster.sendTransform(translation, rotation, time, child, parent)
		rate.sleep()

if __name__ == "__main__":
	tf_frame_broadcaster()
	print("Transform frame is broadcasting")

File: rosbasic/src/tf_frame_broadcaster.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>

using namespace std;

int main(int argc,char **argv){
  ros::init(argc,argv,"broadcaster_node");
  ros::NodeHandle nh;

  tf::TransformBroadcaster br;
  tf::Transform transform;

  ros::Rate rate(10.0);

  while(ros::ok()){
    transform.setOrigin(tf::Vector3(7.0,5.0,0.0));
    transform.setRotation(tf::Quaternion(0,0,0,1));
    ros::Time time = ros::Time::now();
    string parent_frame = "/parent";
    string child_frame = "/child";
    br.sendTransform(tf::StampedTransform(transform,time,parent_frame,child_frame));
    rate.sleep();
  }
  return 0;
};
